src/main/java/org/jenkinsci/plugins/workflow/cps/CpsWhitelist.java

```diff
@@ -103,6 +104,9 @@ public boolean permitsStaticMethod(Method method, Object[] args) {
             String driverTo = "org.codehaus.groovy.runtime.";
             if (cn.startsWith(driverFrom)) {
                 try {
+                    if (Modifier.isPrivate(method.getModifiers())) {
+                        return true;
```

Comment:
To be clear, my reluctance to accept the current patch is that it is not clear that all `private static` methods in `Cps*` are in fact safe. Just because we decline to whitelist the `public` methods which are supposed to delegate to them does not block a script from calling them directly, since Groovy allows access to `private` methods.